### ISSUE TITLE
Show more verbose registration errors

### ISSUE THREAD

**Issue by kantai** [blockstack/blockstack-browser#628]
When a registration error occurs due to a low balance (this may happen if the initial price lookup and then the registration checks arrive at slightly different prices), core will respond with a 500 HTTP Error and the following json:

```
{
    "error": "register failed.\nAddress XYZF does not have enough balance (need 412592, have 300655)."
}
```

However, browser doesn't display an informational error to the user.

I realize that this is not a terribly easy to parse error message, so I could support the following instead:

```
{
    "error": {
        "message": "register failed.",
        "reason": {
            "address": "39cRw6iMETgz9vpqGfQzCcfdcujWat9rPU",
            "have": 300655,
            "message": "low balance",
            "message_verbose": "Address does not have enough balance",
            "needed": 411736
        }
    }
}
```

I'd put this on the head of `rc-0.14.3`, so browser would need to package with a new version of core.

**Comment by shea256** [comment]
It'd be awesome to model errors after those in one of the more well-designed systems in the space - Stripe.

Here's how they do errors:

https://stripe.com/docs/relay/apps-error-guide

Here's a good example:

```
{
  "error": {
    "code": "out_of_inventory",
    "param": "items[0]",
    "message": "This SKU cannot be used to create an order: sku_80jcxrvoPNd57O is out of stock",
    "type": "invalid_request_error"
  }
}
```

Specifically, the unique "code" is key. It makes it very easy for a developer to classify the error and know how to handle it.

I like how you have the specific balance and required balance listed. Maybe that can go in a error-specific context object.

@larrysalibra probably has more thoughts on this as well.

**Comment by kantai** [comment]
Yeah -- that error model is the basis for the core issue here (https://github.com/blockstack/blockstack-core/issues/422) and in general, I like the approach Stripe takes. However, I don't like that it requires string parsing to get at the semantic bits (here, that would be the necessary satoshis and the current satoshis). So we could have a flatter error message, but one that still had some separate fields for data?

```
{
    "error": {
        "message": "Address does not have enough balance to complete a registration.",
        "code" : "low_balance",
        "details": {
            "address": "39XYZF",
            "has_satoshi": 300655,
            "needed_satoshi": 411736
        },
        "type": "operation_safety_check_failed"
    }
}
```
